pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -621,39 +618,36 @@ abstract contract LinearPool is ILinearPool, IGeneralPool, IRateProvider, BasePo
         _require(upperTarget % _TARGET_SCALING == 0, Errors.FRACTIONAL_TARGET);
         _require(lowerTarget % _TARGET_SCALING == 0, Errors.FRACTIONAL_TARGET);
 
-        _setMiscData(
+        _poolState =
             WordCodec.encodeUint(lowerTarget / _TARGET_SCALING, _LOWER_TARGET_OFFSET, _TARGET_BITS) |
-                WordCodec.encodeUint(upperTarget / _TARGET_SCALING, _UPPER_TARGET_OFFSET, _TARGET_BITS)
-        );
+            WordCodec.encodeUint(upperTarget / _TARGET_SCALING, _UPPER_TARGET_OFFSET, _TARGET_BITS);
```

Comment:
This is wrong - you're overriding the state entirely by only writing the new targets. You should instead use `insert` (which is what `setMiscData()` does).